### PR TITLE
Pipeline and test changes for segment wal replication

### PR DESCRIPTION
**Pull request by asubramanya**
This PR contains the following commits:
* Pipeline changes to enable testing of segment wal replication feature 
* Fix the walrep generate_ao_xlog/generate_aoco_xlog tests so that they do not hang.

### CODE REVIEW

File: concourse/tasks/ic_gpdb.yml [hunk 1]
```diff
@@ -9,5 +9,6 @@ params:
   MAKE_TEST_COMMAND: ""
   BLDWRAP_POSTGRES_CONF_ADDONS: ""
   TEST_OS: ""
+  CONFIGURE_FLAGS: ""
```
Comment (cjcjameson) [inline concourse/tasks/ic_gpdb.yml]: right... the ic_gpdb.bash script is actually running configure under the hood...

File: src/test/walrep/gplibpq.c [hunk 2]
```diff
@@ -390,12 +393,48 @@ check_ao_record_present(unsigned char type, char *buf, Oid spc_node,
...
-			xl_ao_insert *xlaoinsert = (xl_ao_insert *)XLogRecGetData(xlrec);
+			if (hdr->xlp_info & XLP_FIRST_IS_CONTRECORD)
+			{
+				contrecord = (XLogContRecord *)((char *)hdr + XLogPageHeaderSize(hdr));
+				elog(DEBUG1, "remaining length of record = %u", contrecord->xl_rem_len);
+				i += MAXALIGN(XLogPageHeaderSize(hdr) + SizeOfXLogContRecord +
+							  contrecord->xl_rem_len);
+			}
+			else
+				i += XLogPageHeaderSize(hdr);
```
Comment (asimrp) [inline src/test/walrep/gplibpq.c]: Please add a comment explaining this case.

File: src/test/walrep/gplibpq.c [hunk 3]
```diff
@@ -390,12 +393,48 @@ check_ao_record_present(unsigned char type, char *buf, Oid spc_node,
...
+		}
+		else if (xlrec->xl_rmid == RM_APPEND_ONLY_ID)
+		{
+			xl_ao_insert *xlaoinsert = palloc0(xlrec->xl_tot_len);
+
+			if (xlrec->xl_tot_len > avail_in_block)
+			{
```
Comment (asimrp) [inline src/test/walrep/gplibpq.c]: This case is when an AO xlog record is split.  A comment indicating that will be helpful.

File: src/test/walrep/gplibpq.c [hunk 4]
```diff
@@ -419,6 +458,13 @@ check_ao_record_present(unsigned char type, char *buf, Oid spc_node,
 					 xlaoinsert->offset, xlrec->xl_len - SizeOfAOInsert);
 			}
 		}
+		else
+		{
+			if (xlrec->xl_tot_len > avail_in_block || avail_in_block < SizeOfXLogRecord)
```
Comment (asimrp) [inline src/test/walrep/gplibpq.c]: Please add a comment explaining second part of `||` in this condition.

**Comment by asubramanya** [comment]
Thanks for the review. Pushed.

**Pull request merged**